### PR TITLE
Create base class for dashboard layout to remove some duplicating code

### PR DESCRIPTION
**Pull request by caradong**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [ ] Include a change request file using `$ npm run change`  Done

#### Description of changes

(give an overview)

#### Focus areas to test

(optional)

### CODE REVIEW

File: packages/dashboard/src/components/DashboardGridLayout/DashboardGridLayout.types.ts [hunk 1]
```diff
@@ -141,6 +139,15 @@ export interface IDashboardGridSectionLayoutProps extends IDashboardGridLayoutPr
    * @default false
    */
   isCollapsible?: boolean;
+  /**
+   * Callback so you can save the layout.
+   */
```
Comment (Raghurk) [inline packages/dashboard/src/components/DashboardGridLayout/DashboardGridLayout.types.ts]: NIT: extra line

File: packages/dashboard/src/components/DashboardGridLayout/DashboardGridLayoutBase.tsx [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+
+export interface IDashboardGridLayoutBaseProps extends IDashboardGridLayoutProps {
+  createRGLLayouts?(dashBoardLayout: DashboardGridBreakpointLayouts): Layouts;
+  onLayoutChange?(currentLayout: Layout[], allLayouts: Layouts): void;
+}
+
+export class DashboardGridLayoutBase extends React.Component<IDashboardGridLayoutBaseProps, {}> {
+  /** The default props used for React-Grid-Layout */
+  public static defaultProps: Partial<IDashboardGridLayoutBaseProps> = {
+    rowHeight: 50,
```
Comment (Raghurk) [inline packages/dashboard/src/components/DashboardGridLayout/DashboardGridLayoutBase.tsx]: >rowHeight: 50, [](start = 4, length = 14)

@samiyaakhtar  can you please take a look at this number?

File: packages/dashboard/src/components/DashboardGridLayout/DashboardGridSectionLayout.tsx [hunk 3]
```diff
@@ -17,30 +18,7 @@ require('style-loader!css-loader!react-grid-layout/css/styles.css');
 require('style-loader!css-loader!react-resizable/css/styles.css');
 require('style-loader!css-loader!./DashboardGridLayout.css');
```
Comment (Raghurk) [inline packages/dashboard/src/components/DashboardGridLayout/DashboardGridSectionLayout.tsx]: do we need this here?

File: packages/dashboard/src/components/DashboardGridLayout/DashboardGridLayoutBase.tsx (hunk 2 above)
Comment (samiyaakhtar) [reply packages/dashboard/src/components/DashboardGridLayout/DashboardGridLayoutBase.tsx]: Yes, we can either change it here or pass a prop from M365.

Comment (caradong) [reply packages/dashboard/src/components/DashboardGridLayout/DashboardGridLayoutBase.tsx]: Talked offline and I am making this 56 

---
In reply to: [216083082](https://github.com/OfficeDev/office-ui-fabric-react/pull/6285#discussion_r216083082) [](ancestors = 216083082)

File: packages/dashboard/src/components/DashboardGridLayout/DashboardGridLayout.types.ts (hunk 1 above)
Comment (caradong) [reply packages/dashboard/src/components/DashboardGridLayout/DashboardGridLayout.types.ts]: resolved

---
In reply to: [216082115](https://github.com/OfficeDev/office-ui-fabric-react/pull/6285#discussion_r216082115) [](ancestors = 216082115)

File: packages/dashboard/src/components/DashboardGridLayout/DashboardGridSectionLayout.tsx (hunk 3 above)
Comment (caradong) [reply packages/dashboard/src/components/DashboardGridLayout/DashboardGridSectionLayout.tsx]: We dont. It is used in the base class. It is not necessary here. Good catch

---
In reply to: [216083249](https://github.com/OfficeDev/office-ui-fabric-react/pull/6285#discussion_r216083249) [](ancestors = 216083249)

**Pull request merged**